### PR TITLE
feat: add odd number check #755

### PR DESCRIPTION
**Pull request by asuzukosi**
Added Odd number check to make sure that scale factor is an odd number before proceeding with the rest of the function.

### CODE REVIEW

File: mitiq/zne/scaling/folding.py [hunk 1]
```diff
@@ -277,6 +277,12 @@ def fold_all(
         raise ValueError(
             f"Requires scale_factor >= 1 but scale_factor = {scale_factor}."
         )
+        
+    if scale_factor % 2 == 0:
```
Comment (crazy4pi314) [inline mitiq/zne/scaling/folding.py:281]: The `scale_factor` parameter seems to be a float, will the mod operator be sufficient here?

**Review by crazy4pi314 — COMMENTED** [review]
Thanks for picking this up!

**Comment by crazy4pi314** [comment]
@rmlarose @andreamari Is this the desired behavior when the scale factor is odd?

**Comment by rmlarose** [comment]
We don't want to raise an error (apologies to @asuzukosi) and the desired behavior for non odd integer scale factors is still up for discussion (https://github.com/unitaryfund/mitiq/issues/755#issuecomment-871243276). I leave the final decision to @andreamari.

**Comment by andreamari** [comment]
I agree with the comment by @crazy4pi314 . We cannot use the `%` operation since `scale_factor` is a float.

Anyway, based on the last comments in the discussion of #755, I am closing this PR.

Thanks anyway to @asuzukosi for your attempt!

**Pull request closed without merging**